app/uk/gov/hmrc/agentoverseasapplicationfrontend/controllers/auth/AgentAffinityNoEnrolmentAuthAction.scala

```
@@ -30,8 +34,34 @@ class AgentAffinityNoEnrolmentAuthActionImpl @Inject()(
     authorised(AuthProviders(GovernmentGateway) and AffinityGroup.Agent)
       .retrieve(credentials and authorisedEnrolments) {
         case creds ~ enrolments =>
-          if (!isEnrolledForHmrcAsAgent(enrolments)) block(CredentialRequest(creds.providerId, request))
-          else Future.successful(Forbidden)
+          if (!isEnrolledForHmrcAsAgent(enrolments)) {
+
+            sessionStoreService.fetchAgentSession.flatMap {
+              case Some(agentSession) => block(CredentialRequest(creds.providerId, request, agentSession))
+              case None =>
+                applicationService.getUserActiveApplication.flatMap {
+                  case None => {
+                    initialiseAgentSession
+                    Future.successful(Redirect(routes.ApplicationController.showAntiMoneyLaunderingForm()))
+                  }
+                  case Some(application) if application.status == Pending => {
+                    sessionStoreService.cacheApplicationPendingSession(
+                      ApplicationNotReady(
+                        creds.providerId,
+                        application.tradingName,
+                        application.applicationCreationDate))
+                    Future.successful(Redirect(routes.StartController.applicationStatus))
+                  }
+                  case Some(application) if application.status == Rejected => {
+                    sessionStoreService.cacheLastApplicationIsRejected(true)
+                    initialiseAgentSession
+                    Future.successful(Redirect(routes.StartController.applicationStatus()))
+                  }
+                  case _ =>
+                    Future.successful(NotImplemented) // if is going to continue to create application need to initialiseAgentSession
```

Should throw an exception here which will lead to the generic error page. Please add a todo as well with the comment. I think NotImplemented will lead to play's error page..

---

this is one of the upcoming routing tickets, it is out of scope for this task which was previously set up